cpp/cppsdk/gsdk.cpp

```diff
@@ -316,10 +318,12 @@ namespace Microsoft
                             {
                                 if ((*i).isString())
                                 {
-                                    m_configSettings[i.key().asCString()] = (*i).asCString();
+									tempConfig[i.key().asCString()] = (*i).asCString();
                                 }
                             }
                         }
+
+						m_configSettings = std::unordered_map<std::string, std::string>(tempConfig);
```

Comment:
>m_configSettings = std::unordered_map<std::string, std::string>(tempConfig); [](start = 6, length = 76)

As discussed in person, I think the copy constructor for unordered_map is going to run twice here, once for the creation of the unordered_map from tempConfig and then once to assign that map to m_configSettings. I think we'd be better off if m_configSettings was a reference, rather than an instance...

Reply:
I was wondering the same when I left a similar comment above. But this might be a good candidate for copy elision, no? Thoughts? I'm sorry I admit I don't have full context here, just wanted to discuss and learn about the double copy-constructor pattern I saw here :)

---
In reply to: [377933752](https://github.com/PlayFab/gsdk/pull/43#discussion_r377933752) [](ancestors = 377933752)

Reply:
Yeah, the compiler might be smart enough to avoid the double copy. Still feels like even one copy is unnecessary though.

---
In reply to: [377935683](https://github.com/PlayFab/gsdk/pull/43#discussion_r377935683) [](ancestors = 377935683,377933752)